src/Functional/FirstIndexOf.php

```diff
@@ -37,11 +37,9 @@ function first_index_of($collection, $value)
     InvalidArgumentException::assertCollection($collection, __FUNCTION__, 1);
 
     foreach ($collection as $index => $element) {
-
-        if ($element === $value) {
+        if ($element === (is_callable($value) ? $value($element) : $value)) {
```

Comment:
Let’s either do a big if/else around the whole thing and implement two different loops (ugly but preferred) or assign the result of is_callable() once.